src/main/java/de/zalando/aruha/nakadi/service/subscription/zk/CuratorZkSubscriptionClient.java

```diff
@@ -0,0 +1,356 @@
+        }
+    }
+
+    private String getSubscriptionPath(final String value) {
+        return "/nakadi/subscriptions/" + subscriptionId + value;
+    }
+
+    private String getPartitionPath(final Partition.PartitionKey key) {
+        return getSubscriptionPath("/topics/" + key.topic + "/" + key.partition);
+    }
+
+    @Override
+    public boolean createSubscription() {
+        try {
+            final String statePath = getSubscriptionPath("/state");
+            final Stat stat = curatorFramework.checkExists().forPath(statePath);
+            if (null == stat) {
+                // node not exists.
+                curatorFramework.create().creatingParentsIfNeeded().withMode(CreateMode.PERSISTENT).forPath(statePath);
+                return true;
+            } else {
+                final String state = new String(curatorFramework.getData().forPath(statePath), CHARSET);
+                return !state.equals(STATE_INITIALIZED);
+            }
+        } catch (final Exception e) {
+            exceptionListener.accept(e);
+        }
+        return false;
+    }
+
+    @Override
+    public void fillEmptySubscription(final Map<Partition.PartitionKey, Long> partitionToOffset) {
+        if (null == partitionToOffset) {
+            log.error("Illegal partition offsets, won't create anything");
+            return;
+        }
+        try {
+            log.info("Creating sessions root");
+            if (null != curatorFramework.checkExists().forPath(getSubscriptionPath("/sessions"))) {
+                curatorFramework.delete().guaranteed().deletingChildrenIfNeeded().forPath(getSubscriptionPath("/sessions"));
+            }
+            curatorFramework.create().withMode(CreateMode.PERSISTENT).forPath(getSubscriptionPath("/sessions"));
+
+            log.info("Creating topics");
+            if (null != curatorFramework.checkExists().forPath(getSubscriptionPath("/topics"))) {
+                log.info("deleting topics recursively");
+                curatorFramework.delete().guaranteed().deletingChildrenIfNeeded().forPath(getSubscriptionPath("/topics"));
+            }
+            curatorFramework.create().withMode(CreateMode.PERSISTENT).forPath(getSubscriptionPath("/topics"));
+
+            log.info("Creating partitions");
+            int counter = 0;
+            for (final Map.Entry<Partition.PartitionKey, Long> p : partitionToOffset.entrySet()) {
+                final String partitionPath = getPartitionPath(p.getKey());
+                curatorFramework.create().creatingParentsIfNeeded().withMode(CreateMode.PERSISTENT).forPath(
+                        partitionPath,
+                        serializeNode(null, null, Partition.State.UNASSIGNED));
+                curatorFramework.create().withMode(CreateMode.PERSISTENT).forPath(
+                        partitionPath + "/offset",
+                        (String.valueOf(p.getValue())).getBytes(CHARSET));
+                if (++counter % 100 == 0) {
```

Comment:
This is just crazy :-D
I thought that `++i` is used only on tech interviews to confuse candidates